libraries/serf_helper.rb

```diff
@@ -52,7 +45,12 @@ def get_agent_json
   end
 
   def get_zip_file_path
-    File.join Chef::Config[:file_cache_path], "serf-#{node["serf"]["version"]}_linux_#{node["serf"]["arch"]}.zip"
+    File.join Chef::Config[:file_cache_path], get_binary_filename
+  end
+
+  def get_binary_url
+    node["serf"]["binary_url"] ||
+    File.join(node["serf"]["base_binary_url"], node["serf"]["version"], get_binary_filename)
```

Comment:
@bbaugher this approach was slightly different, but I think it achieves the same behavior. The user has the ability to override the `binary_url` to set a completely different remote source. But if that attr is nil, we'll fall back on the Hashicorp release page as the source.